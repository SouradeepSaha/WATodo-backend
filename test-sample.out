feature 1 output:
INSERT INTO todo_member (email, username_id, password, first_name, last_name) values ('csuser@cs.com', 'myusername', 'sfs3r2', 'Bill', 'Musk')    1 row(s) affected    0.0068 sec

feature 2 output:
SELECT * FROM todo_member WHERE first_name='Gch' AND last_name='viTwabHQ' LIMIT 0, 1000    1 row(s) returned    0.011 sec / 0.000048 sec

feature 3 output:
DELETE FROM todo_task WHERE task_id=3    1 row(s) affected    0.0077 sec

feature 4 output:
UPDATE Task SET task_name='Better Name' WHERE task_id=7;   Query OK, 1 row affected (0.03 sec)  Rows matched: 1  Changed: 1  Warnings: 0
