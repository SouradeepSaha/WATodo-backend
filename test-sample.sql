--feature 1: insert a new member at sign up
INSERT INTO Users (username, email, password, name) VALUES
("johnsmith", "johnsmith@gmail.com", PasS@Wor93v, "John Smith");
 
--feature 2: check whether user with credentials exists at log in
SELECT username FROM Users where username like "andrewdawson178908";

--feature 3: delete tasks
DELETE FROM cs348.Tasks WHERE task_id= "1435";

--feature 4: update the user in
UPDATE cs348.Tasks SET task_name = 'modified task name' WHERE task_id = 1435;
UPDATE cs348.Tasks SET task_name = 'modified task name' WHERE task_id = 1436;

--feature 5: display tasks with specific tag
create view TasksWithTag AS
select TaskTaskId
from TagTasks
where TagTagId = 10;

select * from Tasks, TasksWithTag
where Tasks.Task_id = TasksWithTag.TaskTaskId;

--feature 6: display tasks without any tag
select * from Tasks
where task_id not in (
select TaskTaskId as task_id from TagTasks
) and UserUserId = 103;

