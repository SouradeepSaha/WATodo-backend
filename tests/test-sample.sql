--feature 1: insert a new member at sign up
INSERT INTO Users (username, email, password, name) VALUES
("johnsmith", "johnsmith@gmail.com", PasS@Wor93v, "John Smith");

--feature 2: check whether user with credentials exists at log in
SELECT Verified FROM cs348.Users WHERE EXISTS (
SELECT * FROM Users
WHERE email = “andrewdawson178908”);

SELECT * FROM cs348.Users
WHERE email = “andrewdawson178908” AND password = “hashedPassword”;

--feature 3: delete tasks
DELETE FROM cs348.Tasks WHERE task_id= "1435";

--feature 4: update the user in
UPDATE cs348.Tasks SET task_name = 'modified task name' WHERE task_id = 1435;
UPDATE cs348.Tasks SET task_name = 'modified task name' WHERE task_id = 1436;

--feature 5: display tasks with specific tag
create view TasksWithTag AS
select TaskTaskId
from TagTasks
where TagTagId = 10;

select * from Tasks, TasksWithTag
where Tasks.Task_id = TasksWithTag.TaskTaskId;

--feature 6: display tasks without any tag
select * from Tasks
where task_id not in (
select TaskTaskId as task_id from TagTasks
) and UserUserId = 103;
