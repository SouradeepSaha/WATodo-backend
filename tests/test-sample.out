feature 1 output:
INSERT INTO Users (username, email, password, name) VALUES
("johnsmith", "johnsmith@gmail.com", PasS@Wor93v, "John Smith");
 
1 row(s) affected	0.016 sec

feature 2 output:
SELECT * FROM Users
WHERE email = "elijones662283@gmail.com" AND password = "$2a$08$yPa4CGd1Injn8ebfJjQeoeAWAXqcoh4ggZ79FG5YuH15xZ.ZkEc.m";
+---------+----------------+--------------------------+--------------------------------------------------------------+-----------+----------+------------------+---------------------+---------------------+
| user_id | username       | email                    | password                                                     | name      | verified | verificationCode | createdAt           | updatedAt           |
+---------+----------------+--------------------------+--------------------------------------------------------------+-----------+----------+------------------+---------------------+---------------------+
|      75 | elijones662283 | elijones662283@gmail.com | $2a$08$yPa4CGd1Injn8ebfJjQeoeAWAXqcoh4ggZ79FG5YuH15xZ.ZkEc.m | Eli Jones |        0 |           187332 | 2022-11-27 19:35:10 | 2022-11-27 19:35:10 |
+---------+----------------+--------------------------+--------------------------------------------------------------+-----------+----------+------------------+---------------------+---------------------+
1 row in set (0.03 sec)


feature 3 output:
DELETE FROM cs348.Tasks WHERE task_id= "60";
Query OK, 1 row affected (0.04 sec)
 
1 row(s) affected	0.032 sec

feature 4 output:
UPDATE cs348.Tasks SET task_name = 'modified task name' WHERE task_id = 123;
Query OK, 0 rows affected (0.03 sec)
Rows matched: 0  Changed: 0  Warnings: 0

UPDATE cs348.Tasks SET task_name = 'modified task name' WHERE task_id = 124;
Query OK, 1 row affected (0.04 sec)
Rows matched: 1  Changed: 1  Warnings: 0

feature 5 output:
create view TasksWithTag AS
select TaskTaskId
from TagTasks
where TagTagId = 10;
Query OK, 0 rows affected (0.04 sec)

select * from Tasks, TasksWithTag
where Tasks.Task_id = TasksWithTag.TaskTaskId;
+---------+---------------------+-------------+-------------+---------------------+----------+---------------------+---------------------+------------+------------+
| task_id | task_name           | description | status      | due_date            | priority | createdAt           | updatedAt           | UserUserId | TaskTaskId |
+---------+---------------------+-------------+-------------+---------------------+----------+---------------------+---------------------+------------+------------+
|    1855 | Drawing Tablet      | clever      | Completed   | 2022-12-22 11:59:59 |        5 | 2022-11-28 02:40:21 | 2022-11-28 02:40:23 |       1142 |       1855 |
|    6013 | Meet with Professor | concerned   | Completed   | 2022-12-15 11:59:59 |        5 | 2022-11-28 02:40:28 | 2022-11-28 02:40:30 |       1430 |       6013 |
|    7966 | Go to church        | clumsy      | Not Started | 2022-12-27 11:59:59 |        4 | 2022-11-28 02:40:54 | 2022-11-28 02:40:56 |       9705 |       7966 |
+---------+---------------------+-------------+-------------+---------------------+----------+---------------------+---------------------+------------+------------+
3 rows in set (0.03 sec)

feature 6 output:
select * from Tasks
where task_id not in (
select TaskTaskId as task_id from TagTasks
) and UserUserId = 103;
+---------+-----------------+-------------+-------------+---------------------+----------+---------------------+---------------------+------------+
| task_id | task_name       | description | status      | due_date            | priority | createdAt           | updatedAt           | UserUserId |
+---------+-----------------+-------------+-------------+---------------------+----------+---------------------+---------------------+------------+
|   13208 | Buy new AirPods | curious     | Not Started | 2022-12-08 11:59:59 |        4 | 2022-11-28 02:40:33 | 2022-11-28 02:40:35 |        103 |
|   17054 | Study for Exam  | crazy       | Not Started | 2022-12-03 11:59:59 |        1 | 2022-11-28 02:40:23 | 2022-11-28 02:40:25 |        103 |
+---------+-----------------+-------------+-------------+---------------------+----------+---------------------+---------------------+------------+
2 rows in set (0.03 sec)
